trajectory_follower_wrapper/src/trajectory_follower_wrapper_node.cpp

```diff
@@ -0,0 +1,181 @@
+ */
+#include "trajectory_follower_wrapper/trajectory_follower_wrapper_node.hpp"
+
+namespace trajectory_follower_wrapper
+{
+  namespace std_ph = std::placeholders;
+
+  TrajectoryFollowerWrapperNode::TrajectoryFollowerWrapperNode(const rclcpp::NodeOptions &options)
+      : carma_guidance_plugins::ControlPlugin(options)
+  {
+    // Create initial config
+    config_ = TrajectoryFollowerWrapperConfig();
+    config_.vehicle_response_lag = declare_parameter<double>("vehicle_response_lag", config_.vehicle_response_lag);
+    config_.time_threshold = declare_parameter<double>("time_threshold", config_.time_threshold);
+
+  }
+
+  rcl_interfaces::msg::SetParametersResult TrajectoryFollowerWrapperNode::parameter_update_callback(const std::vector<rclcpp::Parameter> &parameters)
+  {
+    auto error_double = update_params<double>({
+      {"vehicle_response_lag", config_.vehicle_response_lag},
+      {"time_threshold", config_.time_threshold}
+      }, parameters);
+
+    rcl_interfaces::msg::SetParametersResult result;
+
+    result.successful = !error_double;
+
+    return result;
+  }
+
+  carma_ros2_utils::CallbackReturn TrajectoryFollowerWrapperNode::on_configure_plugin()
+  {
+    // Reset config
+    config_ = TrajectoryFollowerWrapperConfig();
+
+    // Load parameters
+    get_parameter<double>("vehicle_response_lag", config_.vehicle_response_lag);
+    get_parameter<double>("time_threshold", config_.time_threshold);
+
+    RCLCPP_INFO_STREAM(rclcpp::get_logger("trajectory_follower_wrapper"), "Loaded Params: " << config_);
+    // Register runtime parameter update callback
+    add_on_set_parameters_callback(std::bind(&TrajectoryFollowerWrapperNode::parameter_update_callback, this, std_ph::_1));
+
+    // Setup subscribers
+    control_cmd_sub_ = create_subscription<autoware_auto_msgs::msg::AckermannControlCommand>("output/control_cmd", rclcpp::QoS{1}.transient_local(),
+                                                              std::bind(&TrajectoryFollowerWrapperNode::ackermann_control_cb, this, std_ph::_1));
+
+    // Setup publishers
+    autoware_traj_pub_ = create_publisher<autoware_auto_msgs::msg::Trajectory>("input/reference_trajectory", 10);
+    autoware_state_pub_ = create_publisher<autoware_auto_msgs::msg::VehicleKinematicState>("input/current_kinematic_state", 10);
+
+    // Setup timers
+    timer_ = create_timer(
+        get_clock(),
+        std::chrono::milliseconds(33), // Spin at 30 Hz per plugin API
+        std::bind(&TrajectoryFollowerWrapperNode::timer_callback, this));
+
+    // Return success if everthing initialized successfully
+    return CallbackReturn::SUCCESS;
+  }
+
+
+  void TrajectoryFollowerWrapperNode::timer_callback()
+  {
+    RCLCPP_DEBUG(get_logger(), "In timer callback");
+
+    if (!current_trajectory_ || !current_pose_ || !current_twist_)
+    {
+      auto autoware_traj_plan = basic_autonomy::waypoint_generation::process_trajectory_plan(current_trajectory_.get(), config_.vehicle_response_lag);
```

Comment:
This needs to be updated
We're currently calling this method using current trajectory if it **doesn't exist**

Reply:
The check here will change to && now